src/Language/Mimsa/Server/Types.hs

```diff
@@ -0,0 +1,8 @@
+module Language.Mimsa.Server.Types where
+
+import Data.IORef
+import Language.Mimsa.Types.AST
+import Language.Mimsa.Types.Store
+
+newtype MimsaEnvironment
+  = MimsaEnvironment {mutableStore :: IORef (Store Annotation)}
```

Comment:
When operating as a server we have one shared `Store` that we loads `StoreExpression`s into as needed. This might not scale - should perhaps consider a big cache for `StoreExpression`s with a TTD as this could end up with every expression ever.